### ISSUE TITLE
Determine beanName just like what `@MockBean` do

### ISSUE THREAD

**Issue by tan9** [pchudzik/springmock#10]
Spring Boot's [`@MockBean`](https://github.com/spring-projects/spring-boot/blob/v1.5.6.RELEASE/spring-boot-test/src/main/java/org/springframework/boot/test/mock/mockito/MockitoPostProcessor.java#L328-L332) determines beanName automatically:
```java
	private String determineBeanName(String[] existingBeans, SpyDefinition definition,
			BeanDefinitionRegistry registry) {
		if (StringUtils.hasText(definition.getName())) {
			return definition.getName();
		}
		if (existingBeans.length == 1) {
			return existingBeans[0];
		}
		return determinePrimaryCandidate(registry, existingBeans,
				definition.getTypeToSpy());
	}
```

So that the mock will automatically override existing beans without raising  `NoUniqueBeanDefinitionException` like:
> No qualifying bean of type 'com.cht.commons.security.login.UserRegistry' available: expected single matching bean but found 2: userRegistry,userRegistryMock

**Comment by pchudzik** [comment]
I will look into it, and figure something out to make it work more like in spring.

Right now bean name generation is based only on the field name.
Until it's fixed you can set mock name using name() attribute on @AutowiredMock and @AutowiredSpy
Or you can simply name your field userRegistry it will replace original userService with the mock.

**Comment by pchudzik** [comment]
So something like [this](https://github.com/pchudzik/springmock/blob/spring_like_naming/infrastructure/src/test/java/com/pchudzik/springmock/infrastructure/test/name/DoubleNameShouldBeResolvedAutomaticallyToReplaceExistingBean.java) will do right?

Will be released in 1.1.1 (sooner than latter ;))

**Comment by pchudzik** [comment]
It is released as 1.1.2
http://search.maven.org/#artifactdetails%7Ccom.pchudzik.springmock%7Cspringmock-spock%7C1.1.2%7C
http://search.maven.org/#artifactdetails%7Ccom.pchudzik.springmock%7Cspringmock-mockito%7C1.1.2%7C

**Comment by tan9** [comment]
@pchudzik I can confirm the issues has been fixed properly.
